### PR TITLE
promote cfn2 to ga

### PR DESCRIPTION
**Pull request by DrFaust92**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudfunction2: promote to `google_cloudfunctions2_function` ga
```

### CODE REVIEW

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/products/cloudfunctions2/terraform.yaml [hunk 1]
```diff
@@ -21,7 +21,6 @@ overrides: !ruby/object:Overrides::ResourceOverrides
     examples:
       - !ruby/object:Provider::Terraform::Examples
         name: "cloudfunctions2_basic"
-        min_version: beta
         primary_resource_id: "function"
         primary_resource_name: "function"
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply mmv1/products/cloudfunctions2/terraform.yaml:25]: sure

**Comment by DrFaust92** [comment]
[c2thorn](https://github.com/c2thorn) gentle reminder 😊

File: mmv1/products/cloudfunctions2/terraform.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (c2thorn) [reply mmv1/products/cloudfunctions2/terraform.yaml]: @DrFaust92 did you intend to remove this line? You've added my suggestion but removed the change that actually promoted to GA

File: mmv1/products/cloudfunctions2/terraform.yaml [hunk 2]
```diff
@@ -21,33 +21,29 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
       - !ruby/object:Provider::Terraform::Examples
         name: "cloudfunctions2_basic"
-        min_version: beta
         primary_resource_id: "function"
-        primary_resource_name: "function"
+        primary_resource_name: "fmt.Sprintf(\"tf-test-function-v2%s\", context[\"random_suffix\"])"
         vars:
+          function: "function-v2"
           bucket_name: "gcf-source"
-          function: "test-function"
```
Comment (DrFaust92) [inline mmv1/products/cloudfunctions2/terraform.yaml:29]: test-test was weird

File: mmv1/templates/terraform/iam/example_config_body/cloudfunctions2_function.tf.erb [hunk 3]
```diff
@@ -1,2 +0,0 @@
-  cloud_function = google_cloudfunctions2_function.%{primary_resource_id}.name
```
Comment (DrFaust92) [inline mmv1/templates/terraform/iam/example_config_body/cloudfunctions2_function.tf.erb:1]: not sure what was the motivation for this but it works fine without it

File: mmv1/products/cloudfunctions2/terraform.yaml [hunk 4]
```diff
@@ -118,8 +112,6 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       eventTrigger.eventFilters: !ruby/object:Overrides::Terraform::PropertyOverride
         is_set: true
     iam_policy: !ruby/object:Api::Resource::IamPolicy
-      min_version: beta
       parent_resource_attribute: 'cloud_function'
       method_name_separator: ':'
       import_format: ["projects/{{project}}/locations/{{location}}/functions/{{cloud_function}}", "{{cloud_function}}"]
-      example_config_body: 'templates/terraform/iam/example_config_body/cloudfunctions2_function.tf.erb'
```
Comment (DrFaust92) [inline mmv1/products/cloudfunctions2/terraform.yaml:125]: if this override is not needed, should this be moved to api.yaml?

Comment (DrFaust92) [reply mmv1/products/cloudfunctions2/terraform.yaml:125]: @c2thorn

**Comment by c2thorn** [comment]
Let me run the suit of CF2 tests in GA, our VCR tests only run beta

File: mmv1/products/cloudfunctions2/terraform.yaml (hunk 4 above)
Comment (c2thorn) [reply mmv1/products/cloudfunctions2/terraform.yaml:125]: No worries, this was related to IAM documentation, but I think it is better without it. Thanks for cleaning up.

File: mmv1/templates/terraform/iam/example_config_body/cloudfunctions2_function.tf.erb (hunk 3 above)
Comment (c2thorn) [reply mmv1/templates/terraform/iam/example_config_body/cloudfunctions2_function.tf.erb:1]: Agreed.

**Review by c2thorn — COMMENTED** [review]
looks good so far, waiting on test results.

**Review by c2thorn — APPROVED** [review]

**Comment by averikitsch** [comment]
@c2thorn Accepting this PR broke our tutorials. Can you revert?

**Comment by c2thorn** [comment]
@averikitsch reverting would completely remove the resource from the GA provider which is a large breaking change. We can discuss offline about the impact?

**Comment by iennae** [comment]
@c2thorn why was the sample changed though? The provider and validator I can understand, but why was the sample modified?

**Comment by c2thorn** [comment]
@iennae The idea was to update all examples/tests to GA to prove it works correctly for both. I think while doing that, the contributor updated some names within the sample to conform with other tests within our provider.

It was my mistake in not understanding/verifying the effect changing the samples could have outside of the provider. I can revert the change for the samples specifically. Since we haven't fully switched away from MM for CGC samples, I'll make the change now

**Comment by bharathkkb** [comment]
@c2thorn @iennae @averikitsch  I fixed this in https://github.com/terraform-google-modules/terraform-docs-samples/blob/switch-ci/cloudfunctions2_full/main.tf. Should we turn off MM and merge https://github.com/terraform-google-modules/terraform-docs-samples/pull/143?

**Comment by c2thorn** [comment]
Thanks @bharathkkb, yes we can go with the approach. I'll turn off generation in MM ASAP. Let's move further discussion off of this pull request, as it no longer seems related.

**Pull request merged**